SayTheirNames/Source/Service/Networking/NetworkRequestor.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  NetworkRequestor.swift
+//  Say Their Names
+//
+//  Created by evilpenguin on 6/3/20.
+//  Copyright © 2020 Franck-Stephane Ndame Mpouli. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+// MARK: - NetworkRequestor
+
+class NetworkRequestor: NSObject, ServiceReferring {
+    let service: Service
+    let concurrentQueue = DispatchQueue(label: "NetworkRequestor", attributes: .concurrent)
+        
+    required init(service: Service) {
+        self.service = service
+    }
+    
+    // MARK: - Public methods
+    
+    public func fetchDecodable<T: Decodable>(_ url: String, completion: @escaping (T?) -> Swift.Void) {
```

Comment:
I think it would provide more flexibility and allow us to handle errors elegantly if we used `Result` instead
```suggestion
    public func fetchDecodable<T: Decodable>(_ url: String, completion: @escaping (Result<T, Error>) -> Swift.Void) {
```

Reply:
Adding